### ISSUE TITLE
mkdocs needs 4 char indents for lists

### ISSUE THREAD

**Issue by jankatins** [rvben/rumdl#522]
### rumdl version

0.1.46

### Installation method

Other

### Bug description

I want to format mkdocs (actually backstage) documentation. With just setting `flavor = "mkdocs"`, rumdl tries to reformat unordered lists to 2 space indented but mkdos needs 4 spaces.

### Steps to reproduce

Ths markdown text 

```md
- text
    - indented
``` 
get reformated to
 
```md
- text
  - indented
```

Error message is `[MD007] Expected 2 spaces for indent depth 1, found 4 [fixed]`

### Expected behavior

It should keep the formatting as is, as it is already correct with 4 spaces.

I need to specify to get it to the right formatting

```
[tool.rumdl.MD007]
# mkdown needs 4 and only "fixed" makes it appear always
indent = 4
style = "fixed"
```

I feel that if mkdocs needs it, I think rumdl should set the default to 4 spaces style = fixedand prevent any changes to that.

### Operating system

Linux

### Additional context

https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown links to https://python-markdown.github.io/ which states:

> Indentation/Tab Length
> The [syntax rules](https://daringfireball.net/projects/markdown/syntax#list) clearly state that when a list item consists of multiple paragraphs, “each subsequent paragraph in a list item must be indented by either 4 spaces or one tab” (emphasis added). However, many implementations do not enforce this rule and allow less than 4 spaces of indentation. The implementers of Python-Markdown consider it a bug to not enforce this rule.
>This applies to any block level elements nested in a list, including paragraphs, sub-lists, blockquotes, code blocks, etc. They must always be indented by at least four spaces (or one tab) for each level of nesting.

**Comment by jankatins** [comment]
There seems to be something wrong with mkdocs enforced spaces:

<img width="107" height="377" alt="Image" src="https://github.com/user-attachments/assets/ca9c839b-55f5-4f12-a41b-4818ef098503" />

I would have expected that the line continuation is always 4 spaces here. If I manually change the indention to 4 spaces in the first line of each item, it gets fixed to 4 chars indention and stays there.

**Comment by rvben** [comment]
Thanks for the report and the detailed context on Python-Markdown's requirements\! Fixed in v0.1.47. When `flavor = "mkdocs"` is set, MD007 now enforces `indent=4` and `style=fixed` as required by Python-Markdown. If a conflicting value is explicitly set (e.g. `indent=2`), a warning is emitted and the value is overridden to 4.

**Comment by jankatins** [comment]
There is still something strange here: If I have a checkbox style list, I get this:

<img width="149" height="173" alt="Image" src="https://github.com/user-attachments/assets/e8cd18b1-790e-465c-85ec-543471cb5aa2" />

And I cannot change this: if I manually add 4 spaces in the continuation lines, it will reformat it again to zero :-( If I remove the checkmark, the 4 spaces are applied.

config is basically this:

```toml

[tool.rumdl]
# We need mkdocs for backstage (can be overridden by per file rules)
flavor = "mkdocs"
exclude = [
 # ...
]
respect-gitignore = true

# Set global line length (used by MD013 and other line-length rules)
line_length = 120

[tool.rumdl.MD013]
# https://rumdl.dev/md013/
# Reflow at semantic line breaks: https://sembr.org/
reflow = true
reflow-mode = "semantic-line-breaks"
code-blocks = false  # Don't check code blocks
tables = false  # Don't check tables
headings = false  # Check headings
paragraphs = true  # Check paragraph/regular text
strict = false  # Exceptions for URLs, etc.
length-mode = "visual"  # How to count line length: "visual", "chars", or "bytes" (default: "visual")
abbreviations = ["Assn", "Univ"]  # Add custom abbreviations for sentence-per-line mode
require-sentence-capital = true  # Require uppercase after periods for sentence detection

[tool.rumdl.MD004]
style = "dash"

[tool.rumdl.MD029]
# https://github.com/rvben/rumdl/blob/main/docs/md029.md
# ordered lists: either all 1. or properly ordered.
style = "one-or-ordered"
```

**Comment by jankatins** [comment]
Move the above follow up into a new story #529
